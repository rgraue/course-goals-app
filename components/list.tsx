import { View, FlatList } from 'react-native';
import { listStyle } from '@styles';
import { Item } from '@components';
import { Goal } from '@interfaces';

export const GoalList = ({goals, handleDelete}) => {

  return (
    <View style={listStyle.container}>
      <FlatList 
        data={goals as Goal[]}
        renderItem={ (goal) => {
          return (
            <Item goal={goal} handlePress={handleDelete} id={goal.item.id}/>
          )
        }}
        keyExtractor={(item) => {
          return item.id
        }}
      />
    </View>
  )
}