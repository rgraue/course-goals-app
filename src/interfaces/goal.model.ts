export interface Goal {
  id: string;
  name: string;
}
