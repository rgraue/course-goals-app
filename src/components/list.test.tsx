import React from 'react';
import { ListRenderItemInfo } from 'react-native';
import { create } from 'react-test-renderer';
import { List } from './list';
import { Goal } from '../interfaces';

const mockGoal: Partial<ListRenderItemInfo<Goal>>[] = [
  {
    item: {
      name: 'test goal name',
      id: '2',
    },
  },
];

test('Goal List component', () => {
  const tree = create(<List goals={mockGoal} handleDelete={jest.fn} />);

  expect(tree).toMatchSnapshot();
});
