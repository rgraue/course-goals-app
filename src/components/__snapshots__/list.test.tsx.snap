// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goal List component 1`] = `
<View
  style={
    Object {
      "flex": 9,
      "padding": 5,
      "width": "85%",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "item": Object {
            "id": "2",
            "name": "test goal name",
          },
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={false}
            >
              <View
                style={
                  Object {
                    "backgroundColor": "grey",
                    "borderRadius": 10,
                    "color": "black",
                    "fontSize": 18,
                    "marginBottom": 10,
                    "padding": 10,
                    "textAlign": "left",
                  }
                }
              >
                <Text />
              </View>
            </View>
            <Modal
              animationType="slide"
              hardwareAccelerated={false}
              transparent={true}
              visible={false}
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
